ingester/src/server/grpc/query.rs

```diff
@@ -363,4 +375,58 @@ mod tests {
             }
         }
     }
+
+    /// Regression test for https://github.com/influxdata/idpe/issues/17408
+    #[tokio::test]
+    #[should_panic(
+        expected = "Invalid argument error: number of columns(1) must match number of fields(2) in schema"
+    )]
+    async fn test_chunks_with_different_schemas() {
+        let schema1 = Arc::new(Schema::new(vec![
+            Field::new("int", DataType::Int32, false),
+            Field::new("float", DataType::Float64, false),
+        ]));
+        let batch1 = RecordBatch::try_new(
+            schema1,
+            vec![
+                Arc::new(Int32Array::from(vec![1, 2, 3])),
+                Arc::new(Float64Array::from(vec![1.1, 2.2, 3.3])),
+            ],
+        )
+        .unwrap();
```

Comment:
It's too late now, but there's a [handy dandy macro](https://github.com/influxdata/influxdb_iox/blob/c5ee3250f731ea770a827e695789aca00aca31f1/ingester/src/test_util.rs#L148-L190) for this:

```suggestion
        let (batch1, schema1) = make_batch!(
            Int32Array("int" => vec![1, 2, 3]),
            Float64Array("float" => vec![1.1, 2.2, 3.3]),
        );
```

Because _man_ is building batches verbose.

Reply:
done